.. change:
    :tags: bug, mssql
    :tickets: 6058

    Fixed regression where a new setinputsizes() API that's available for
    pyodbc was enabled, which is apparently incompatible with pyodbc's
    fast_executemany() mode in the absence of more accurate typing information,
    which as of yet is not fully implemented or tested. The pyodbc dialect and
    connector has been modified so that setinputsizes() is not used at all
    unless the parameter ``use_setinputsizes`` is passed to the dialect, e.g.
    via :func:`_sa.create_engine`, at which point its behavior can be
    customized using the :meth:`.DialectEvents.do_setinputsizes` hook.

    .. seealso::

      :ref:`mssql_pyodbc_setinputsizes`
